doc/adr/0003-standardize-storage-target-layout.md

```diff
@@ -39,38 +39,60 @@ The Path is the string used to point to a dataset.
 A Path string has the following structure
 
 ```
-{prefix}/pangeo-forge/{feedstock_name}/{recipe_name}.{dataset_type}
+{prefix}/pangeo-forge/{subcatalog_name}/{feedstock_name}/{recipe_name}-{release_tag}.{dataset_type}
 ```
 
 This template contains four keys:
 - `prefix`: the root location for the target. This could just be a bucket name,
   or it could be a sub-path within a bucket.
   **Question: should prefix include the protocol, e.g. s3:// ?**
   For now I am assuming no.
-- `feedstock_name`: the name of the repository that holds the feedstock with
-  the recipe that generated the dataset.
+- `subcatalog_name`: [**optional**] the name of a subcatalog to which the dataset belongs. If `subcatalog_name` is not provided, the feedstock will be included in the top-level Pangeo Forge STAC Catalog.
```

Comment:
I agree it would be useful to allow a given dataset to be included in arbitrary catalogs. However, I'm not convinced it needs to be in the storage path name. There are advantages to having the storage path name map closely to feedstocks and recipes. 

Here is an alternative way we could support having recipes appear in different catalogs. Rather than having `subcatalog_name` be part of the dataset path, we could allow datasets to be added to any catalog, via a new keyword in `meta.yaml`, or even an external database. We could even imagine giving users the ability to define their own catalogs and share them with others, e.g. "my favorite datasets". None of this requires us to change the storage path naming conventions.